tests/python/apps/solana.py

```diff
@@ -72,37 +76,63 @@ class SolanaClient:
     def __init__(self, client):
         self._client = client
 
+
     def get_public_key(self, derivation_path : bytes) -> bytes:
         public_key:RAPDU = self._client.exchange(CLA, INS.INS_GET_PUBKEY, P1_NON_CONFIRM, P2_MORE, derivation_path)
         assert len(public_key.data) == PUBLIC_KEY_LENGTH, "'from' public key size incorrect"
         return public_key.data
 
-    def sign_message(self, derivation_path : bytes, message: bytes, accept: bool) -> RAPDU:
-        assert len(message) <= 65535, "Message to sign is too long"
 
+    def split_and_prefix_message(self, derivation_path : bytes, message: bytes) -> List[bytes]:
+        assert len(message) <= 65535, "Message to send is too long"
         header: bytes = _extend_and_serialize_multiple_derivations_paths([derivation_path])
-
         # Check to see if this data needs to be split up and sent in chunks.
         max_size = MAX_CHUNK_SIZE - len(header)
         message_splited = [message[x:x+max_size] for x in range(0, len(message), max_size)]
         # Add the header to every chunk
-        message_splited_prefixed = [header + s for s in message_splited]
+        return [header + s for s in message_splited]
+
+
+    def send_message_batch(self, messages: List[bytes], p1: int) -> RAPDU:
+        for m in messages:
+            self._client.exchange(CLA, INS.INS_SIGN_MESSAGE, p1, P2_MORE | P2_EXTEND, m)
```

Comment:
first batch shouldn't have P2_EXTEND bit set. otherwise the app would try to extend the data buffer with the new data instead of resetting context

Reply:
Nice catch, I didn't test a lot the case where the message has to be split and sent in batches